client/src/redux/sagas/buildSetSaga.js

```diff
@@ -1,16 +1,61 @@
-import { takeEvery, call, put } from 'redux-saga/effects';
-import { BUILD_DATA } from '../constants';
-import { fetchBuildSets } from '../api';
-import { setBuildSets, setBuildSetsError } from '../actions';
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+import { COMMITS } from '../constants';
+import { fetchBuildSets, fetchBuildReq, fetchBuilds } from '../api';
+import {
+ setBuildSets,
+ setBuildSetsError,
+ setBsID,
+ setBuilds
+} from '../actions';
+
+const commitSha = state => state.commits;
+function buildStr(build) {
+ let str = build
+  .map(build => 'buildrequestid__contains=' + build.buildrequestid)
+  .join('&');
+ return str;
+}
+
+function matchSha(commits, buildData) {
```

Comment:
This function is very complicated :)
Here is my version
```suggestion
function getBuildSetReqString(commits, buildData) {
 return commits
  .map(commit => buildData.filter(bd => bd.sourcestamps[0].revision === commit.sha))
  .flat()
  .map(bd => 'buildsetid__contains=' + bd.bsid)
  .join('&');
}
```

Reply:
but here we got the query string directly, but later we will require the merged array for filtering, so i kept a var here(to get global scope) and without writing the same code again